### PR TITLE
Kinesis test

### PR DESCRIPTION
**Pull request by gurbuzali**

### CODE REVIEW

File: kinesis-test/src/main/java/com/hazelcast/jet/tests/kinesis/KinesisTest.java [hunk 1]
```diff
@@ -0,0 +1,217 @@
...
+    public static void main(String[] args) throws Exception {
+        System.setProperty(SDKGlobalConfiguration.AWS_CBOR_DISABLE_SYSTEM_PROPERTY, "true");
+        new KinesisTest().run(args);
+    }
+
+    @Override
+    protected void init(JetInstance client) {
+        awsConfig = new AwsConfig()
+                .withEndpoint(property("endpoint", "http://localhost:4566"))
+                .withRegion(Regions.US_EAST_1.getName())
```
Comment (jbartok) [inline kinesis-test/src/main/java/com/hazelcast/jet/tests/kinesis/KinesisTest.java:73]: This doesn't seem to be configurable via properties.

File: kinesis-test/src/main/java/com/hazelcast/jet/tests/kinesis/KinesisVerificationP.java [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+import java.util.PriorityQueue;
+
+import static com.hazelcast.jet.core.ProcessorMetaSupplier.forceTotalParallelismOne;
+import static com.hazelcast.jet.impl.pipeline.SinkImpl.Type.TOTAL_PARALLELISM_ONE;
+
+public class KinesisVerificationP extends AbstractProcessor {
+
+    public static final String CONSUMED_MESSAGES_MAP_NAME = "KinesisTest_latestCounters";
+
+    private static final int QUEUE_SIZE_LIMIT = 4_000;
```
Comment (olukas) [inline kinesis-test/src/main/java/com/hazelcast/jet/tests/kinesis/KinesisVerificationP.java:36]: What `sleepMsBetweenItem` we will use? Currently `DEFAULT_SLEEP_MS_BETWEEN_ITEM = 1` which means that this queue limit is really small. However I guess we will use higher `sleepMsBetweenItem` than `1`, right?

File: kinesis-test/pom.xml [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+    </properties>
+
+    <dependencies>
+        <dependency>
+            <groupId>com.hazelcast.jet</groupId>
+            <artifactId>hazelcast-jet-kinesis</artifactId>
+            <version>${jet.version}</version>
+        </dependency>
+        <dependency>
+            <groupId>com.google.guava</groupId>
```
[the remark this answers is not in the document]
Comment (gurbuzali) [reply kinesis-test/pom.xml:27]: copt/paste leftover

File: kinesis-test/src/main/java/com/hazelcast/jet/tests/kinesis/KinesisTest.java (hunk 1 above)
Comment (gurbuzali) [reply kinesis-test/src/main/java/com/hazelcast/jet/tests/kinesis/KinesisTest.java:73]: I've removed it intentionally, we run soak-tests on `us_east_1` and it is hardcoded. I thought it'll be the same for kinesis so removed the parameter.

File: kinesis-test/src/main/java/com/hazelcast/jet/tests/kinesis/KinesisVerificationP.java (hunk 2 above)
Comment (gurbuzali) [reply kinesis-test/src/main/java/com/hazelcast/jet/tests/kinesis/KinesisVerificationP.java:36]: Yes, I was thinking 20ms, let me change the default to that one too.

File: kinesis-test/src/main/java/com/hazelcast/jet/tests/kinesis/KinesisTest.java [hunk 4]
```diff
@@ -0,0 +1,217 @@
...
+        waitForJobStatus(writeJob, RUNNING);
+
+        int cycles = 0;
+        int latestCycle = 0;
+        int latestCheckedCount = 0;
+        long begin = System.currentTimeMillis();
+        while (System.currentTimeMillis() - begin < durationInMillis) {
+            if (getJobStatusWithRetry(readJob) == FAILED) {
+                writeJob.cancel();
+                writeJob.join();
```
[the remark this answers is not in the document]
Comment (gurbuzali) [reply kinesis-test/src/main/java/com/hazelcast/jet/tests/kinesis/KinesisTest.java:117]: yes, when we observe the failed status, we should just cancel the other job and join to the failed job, no need to join to the other job.

File: kinesis-test/src/main/java/com/hazelcast/jet/tests/kinesis/KinesisTest.java [hunk 5]
```diff
@@ -0,0 +1,217 @@
...
+        sleepMsBetweenItem = propertyInt("sleepMsBetweenItem", DEFAULT_SLEEP_MS_BETWEEN_ITEM);
+        snapshotIntervalMs = propertyInt("snapshotIntervalMs", DEFAULT_SNAPSHOT_INTERVAL);
+
+        helper = new Helper(awsConfig.buildClient(), logger);
+    }
+
+    @Override
+    protected void teardown(Throwable t) {
+        for (String streamName : streamNames) {
+            helper.deleteStream(streamName);
```
[the remark this answers is not in the document]
Comment (gurbuzali) [reply kinesis-test/src/main/java/com/hazelcast/jet/tests/kinesis/KinesisTest.java:88]: 👍

File: kinesis-test/src/main/java/com/hazelcast/jet/tests/kinesis/KinesisTest.java [hunk 6]
```diff
@@ -85,7 +85,11 @@ protected void init(JetInstance client) {
     @Override
     protected void teardown(Throwable t) {
         for (String streamName : streamNames) {
-            helper.deleteStream(streamName);
+            try {
+                helper.deleteStream(streamName);
+            } catch (Exception e) {
+                logger.severe("Exception when deleting stream: " + streamName, e);
```
Comment (olukas) [inline kinesis-test/src/main/java/com/hazelcast/jet/tests/kinesis/KinesisTest.java:91]: I would suggest to fail the test after all streams tried to be deleted and any of them was not deleted. In that case we need to delete them manually. Test failure should inform us about that, I think we can miss it if it is only logged. However part of test failure should be an information that this particular exception does not mean failed soak test it just means that some streams cannot be deleted and should be deleted manually. We should also put there an info that the rest of test log should be checked whether there is also some test failure.

**Comment by gurbuzali** [comment]
should be merged after https://github.com/hazelcast/hazelcast-jet/pull/2656

**Review by olukas — APPROVED** [review]

**Pull request merged**